.github/workflows/linux-tests.yml

```diff
@@ -0,0 +1,125 @@
+name: GitHub Actions CI - (Linux)
+on:
+  pull_request:
+    paths-ignore:
+      # Slow running processes that will be built on the self-hosted runner
+      - "Formula/bison.rb"
+      - "Formula/bzip2.rb"
+      - "Formula/curl.rb"
+      - "Formula/expat.rb"
+      - "Formula/flex.rb"
+      - "Formula/libffi.rb"
+      - "Formula/libxml2.rb"
+      - "Formula/libxslt.rb"
+      - "Formula/libx11.rb"
+      - "Formula/m4.rb"
+      - "Formula/ncurses.rb"
+      - "Formula/perl.rb"
+      - "Formula/python@3.9.rb"
+      - "Formula/sqlite.rb"
+      - "Formula/zlib.rb"
+      # Will be built in the linux-tests-wheezy workflow
+      - "Formula/patchelf.rb"
+      - "Formula/binutils.rb"
+jobs:
+  check_labels_linux:
+    runs-on: ubuntu-latest
+    outputs:
+      run-tests: ${{ steps.check-labels.outputs.result }}
+    steps:
+      - name: Check for CI-syntax-only label
+        id: check-labels
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { data: { labels: labels } } = await github.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            })
+            if (!labels.map(label => label.name).includes('CI-syntax-only')) {
+              console.log('No CI-syntax-only label found. Running tests.')
+              return true
+            }
+            console.log('CI-syntax-only label found. Skipping tests.')
+
+  tests_linux:
+    needs: check_labels_linux
+    if: needs.check_labels_linux.outputs.run-tests
+    runs-on: ubuntu-latest
+    container:
+      image: homebrew/ubuntu16.04:master
+    env:
+      HOMEBREW_CORE_GIT_REMOTE: ${{github.event.repository.html_url}}
```

Comment:
Since we don't run the doctor step, this env variable and HOMEBREW_FORCE_HOMEBREW_ON_LINUX might not be needed because setup-homebrew already configures the right tap for us. Speaking this after creating the dispatch workflow, though I'm not fully sure.